### PR TITLE
Improved drive API

### PR DESCRIPTION
**Pull request by acaldas**

### CODE REVIEW

**Review by ryanwolhuter — APPROVED** [review]
looks good, probably worth it to wait for frank's approval too though

**Comment by acaldas** [comment]
> looks good, probably worth it to wait for frank's approval too though

This is needed for the Read Mode feature, which is meant for tomorrow's release. I'll go through the changes with him when he comes back.

**Pull request merged**
